src/monattack.cpp

```diff
@@ -4647,3 +4648,19 @@ bool mattack::dodge_check( monster *z, Creature *target ) {
     }
     return false;
 }
+
+void mattack::lay_egg( monster *z )
+{
+    monster &critter = *z;
+    const furn_t &furn = g->m.furn( critter.pos() ).obj();
+
+    if( !critter.is_dead() && critter.has_flag( MF_EGGLAYING ) && furn.has_flag( TFLAG_NEST ) &&
```

Comment:
Skip the egglaying check here - a monster that has `lay_egg` attack must be egglaying.
Would be clearer if you renamed it to `MF_SEEKS_NEST` or something like that.